clang/Type.d

```diff
@@ -93,31 +117,55 @@ struct Type
         return isObjCIdType || isObjCClassType || isObjCSelType;
     }
 
-    @property bool isWideCharType ()
+    @property bool isPointer ()
     {
-        with (CXTypeKind)
-            return kind == CXType_WChar;
+        return kind == CXTypeKind.CXType_Pointer;
     }
 
-    @property bool isConst ()
+    @property bool isTypedef ()
     {
-        return clang_isConstQualifiedType(cx) == 1;
+        return kind == CXTypeKind.CXType_Typedef;
     }
 
-    @property bool isExposed ()
+    @property bool isValid ()
     {
-        return kind != CXTypeKind.CXType_Unexposed;
+        return kind != CXTypeKind.CXType_Invalid;
     }
 
-    @property bool isAnonymous ()
+    @property bool isWideCharType ()
+    {
+        return kind == CXTypeKind.CXType_WChar;
+    }
+
+    @property Type canonical()
+    {
+        if (isClang)
+            return Type(clang_getCanonicalType(cx));
+        else
+            return Type.init;
+    }
+
+    @property Type pointee()
     {
-        return spelling.isEmpty;
+        if (pointee_)
+        {
+            return *pointee_;
+        }
+        else
+        {
+            if (isClang)
+                return Type(clang_getPointeeType(cx));
+            else
+                return Type.init;
+        }
     }
 
     @property Cursor declaration ()
     {
-        auto r = clang_getTypeDeclaration(cx);
-        return Cursor(r);
+        if (isClang)
```

Comment:
Isn't it better to us an `assert`, because you're not really supposed to call this method unless the type is coming from Clang?